### PR TITLE
Add Date-time stamp in NuGet package versioning for appropriate ordering of the packages

### PR DESCRIPTION
**Pull request by shahasad**
**Description**: Describe your changes.

**Motivation and Context**
- Why is this change required? What problem does it solve?
Current nightly dev packages are versioned with git commit hash which does not have incremental ordering. So added build date-time in the version so that in the NuGet feed, latest version packages show up at the top 
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: csharp/src/Microsoft.ML.OnnxRuntime/Microsoft.ML.OnnxRuntime.csproj [hunk 1]
```diff
@@ -200,14 +200,17 @@
     </Exec>
 
     <PropertyGroup>
+      <CurrentDate>$([System.DateTime]::UtcNow.ToString(yyyyMMdd))</CurrentDate>
+      <CurrentTime>$([System.DateTime]::UtcNow.ToString(hhmm))</CurrentTime>
       <PackageVersion>@(MajorVersionNumber)</PackageVersion>
       <Version>$(PackageVersion)</Version>
-      <PackageVersion Condition="'$(IsReleaseBuild)'!='true'">$(PackageVersion)-dev-$(GitCommitHashShort)</PackageVersion>
+      <PackageVersion Condition="'$(IsReleaseBuild)'!='true'">$(PackageVersion)-dev-$(CurrentDate)-$(CurrentTime)-$(GitCommitHashShort)</PackageVersion>
```
Comment (jignparm) [inline csharp/src/Microsoft.ML.OnnxRuntime/Microsoft.ML.OnnxRuntime.csproj]: Does Visual Studio still recognize this format as a "dev" package? I'm assuming the sort order in VS will be improved where are are multiple dev packages available in the source location.

File: tools/ci_build/github/azure-pipelines/nuget/templates/upload-binary-sizes-from-nuget-package.yml [hunk 2]
```diff
@@ -33,7 +34,7 @@ steps:
         echo processing %%~ni.nupkg
         copy %%~ni.nupkg %%~ni.zip
         echo copied to zip
-        echo listing lib files in the zip
+        echo listing lib files in the zip
```
Comment (jignparm) [inline tools/ci_build/github/azure-pipelines/nuget/templates/upload-binary-sizes-from-nuget-package.yml]: nit: extra space

File: csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests.Capi/runtest-gpu.bat [hunk 3]
```diff
@@ -3,19 +3,9 @@ REM Licensed under the MIT License.
...
-    FOR /F "tokens=* USEBACKQ" %%F IN (`git rev-parse --short HEAD`) DO (
-        set VersionSuffix=-dev-%%F
-    )
-)
-
-set CurrentOnnxRuntimeVersion=%MajorVersionNumber%%VersionSuffix%
-
 @echo %CurrentOnnxRuntimeVersion%
 
 pushd test\Microsoft.ML.OnnxRuntime.EndToEndTests.Capi
```
Comment (jignparm) [inline csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests.Capi/runtest-gpu.bat]: The C# tests for GPU (in the CI build) seem to be getting skipped, even though TESTONGPU flag is set in **gpu-esrp-pipeline.yml**. May not be related to this change, but I believe they should be running.

```
[xUnit.net 00:00:02.19]     Microsoft.ML.OnnxRuntime.Tests.InferenceTest.TestGpu [SKIP]
```

File: csharp/src/Microsoft.ML.OnnxRuntime/Microsoft.ML.OnnxRuntime.csproj (hunk 1 above)
Comment (shahasad) [reply csharp/src/Microsoft.ML.OnnxRuntime/Microsoft.ML.OnnxRuntime.csproj]: yes, this is still recognized as dev package. VS or NuGet repo orders the text after the main n.n.n version number lexicographically, so, having an equal length string with datetime, will bring the latest on the top.

File: csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests.Capi/runtest-gpu.bat (hunk 3 above)
Comment (shahasad) [reply csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests.Capi/runtest-gpu.bat]: Thanks for noting this. Let me look into that issue as a separate one, This seems to be happening for a while before this change.

**Pull request merged**